src/pkg/p2p/preheat/provider/kraken.go

```diff
@@ -0,0 +1,131 @@
+package provider
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/docker/distribution/manifest/schema2"
+	common_models "github.com/goharbor/harbor/src/common/models"
+	"github.com/goharbor/harbor/src/common/utils"
+	"github.com/goharbor/harbor/src/pkg/p2p/preheat/dao/models"
+	"github.com/goharbor/harbor/src/pkg/p2p/preheat/provider/auth"
+	"github.com/goharbor/harbor/src/pkg/p2p/preheat/provider/client"
+	"github.com/goharbor/harbor/src/pkg/registry"
+	"time"
+)
+
+const (
+	krakenHealthPath  = "/health"
+	krakenPreheatPath = "/registry/notifications"
+)
+
+// KrakenDriver implements the provider driver interface for Uber kraken.
+// More details, please refer to https://github.com/uber/kraken
+type KrakenDriver struct {
+	instance *models.Metadata
+}
+
+// Self implements @Driver.Self.
+func (kd *KrakenDriver) Self() *Metadata {
+	return &Metadata{
+		ID:          "kraken",
+		Name:        "Kraken",
+		Icon:        "https://github.com/uber/kraken/blob/master/assets/kraken-logo-color.svg",
+		Version:     "0.1.3",
+		Source:      "https://github.com/uber/kraken",
+		Maintainers: []string{"mmpei/peimingming@corp.netease.com"},
+		AuthMode:    auth.AuthModeNone,
+	}
+}
+
+// GetHealth implements @Driver.GetHealth.
+func (kd *KrakenDriver) GetHealth() (*DriverStatus, error) {
+	if kd.instance == nil {
+		return nil, errors.New("missing instance metadata")
+	}
+
+	url := fmt.Sprintf("%s%s", strings.TrimSuffix(kd.instance.Endpoint, "/"), krakenHealthPath)
+	_, err := client.DefaultHTTPClient.Get(url, kd.getCred(), nil, nil)
+	if err != nil {
+		// Unhealthy
+		return nil, err
```

Comment:
Why not `DriverStatusUnHealthy` ?

Reply:
it seams a uniform format. error will be logged outside.